### PR TITLE
Updated Document Upload and details

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

**Comment by devinleighsmith** [comment]
@FuriousLlama there are some minor variations compared to the mockup that I see in https://jira.th.gov.bc.ca/browse/PSP-7338

Can you please confirm that what you have in this story is approved by Ana, and if so, get here to update the mockups so we don't get test failures.

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+}) => {
+  const validDocumentExtensions: string = validExtensions.map(x => `.${x}`).join(',');
+
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleFileInput = (changeEvent: ChangeEvent<HTMLInputElement>) => {
+    // handle validations
+    if (changeEvent.target !== null) {
+      var target = changeEvent.target;
+      if (target.files !== null) {
```
Comment (devinleighsmith) [inline source/frontend/src/components/common/form/FileDragAndDrop.tsx:25]: target.files?.length !== null & target.files.length >= 1
(or similar)

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    if (files && files.length > 0) {
+      onSelectFile(files[0]);
+      event.dataTransfer.clearData();
+    }
+
+    setIsDragging(false);
+  };
+
+  const shortenString = (text: string, maxlenght: number, terminator: string = '...'): string => {
```
Comment (devinleighsmith) [inline source/frontend/src/components/common/form/FileDragAndDrop.tsx:66]: maxlenght -> maxLength

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+              onChange={handleFileInput}
+              className=""
+            />
+          </StyledUploadLabel>
+        </StyledContent>
+      </DragDropZone>
+      {selectedFile !== null && (
+        <StyledSelectedFile>
+          <SectionField label="File to add" labelWidth="3">
+            {shortenString(selectedFile.name || '', 20)}
```
Comment (devinleighsmith) [inline source/frontend/src/components/common/form/FileDragAndDrop.tsx:102]: maybe could use the OverflowTip component instead?

**Review by devinleighsmith — COMMENTED** [review]
was this tested on multiple browsers? I do wonder if there could be any variations.

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+import { SectionField } from '../Section/SectionField';
+
+interface IFileDragAndDropProps {
+  onSelectFile: (file: File | null) => void;
+  selectedFile: File | null;
+  validExtensions: string[];
+}
+
+const FileDragAndDrop: FunctionComponent<React.PropsWithChildren<IFileDragAndDropProps>> = ({
```
Comment (devinleighsmith) [inline source/frontend/src/components/common/form/FileDragAndDrop.tsx:12]: The test coverage on this isn't great. Unit testing this completely may not be realistic but is there anything we can do here?

File: source/frontend/src/features/documents/documentDetail/DocumentDetailView.tsx [hunk 5]
```diff
@@ -40,31 +34,32 @@ export const DocumentDetailView: React.FunctionComponent<
...
-                      props.setIsEditable(true);
-                    }}
-                  >
-                    <FaEdit />
-                  </LinkButton>
-                </Col>
-              )}
-            </Row>
+
+          <Section
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentDetail/DocumentDetailView.tsx:38]: This seems duplicated from the DocumentDetailHeader

File: source/frontend/src/features/documents/documentUpload/DocumentUploadContainer.tsx [hunk 6]
```diff
@@ -131,8 +148,21 @@ export const DocumentUploadContainer: React.FunctionComponent<
...
+  };
+
+  useImperativeHandle(ref, () => ({
+    uploadDocument() {
+      formikRef.current?.submitForm();
+    },
+  }));
+
+  const onDocumentSelected = () => {
+    debugger;
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentUpload/DocumentUploadContainer.tsx:164]: thinking this was not intentional.

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx (hunk 1 above)
Comment (FuriousLlama) [reply source/frontend/src/components/common/form/FileDragAndDrop.tsx:25]: sure, added the check for at least one file

File: source/frontend/src/components/common/form/FileDragAndDrop.tsx (hunk 3 above)
Comment (FuriousLlama) [reply source/frontend/src/components/common/form/FileDragAndDrop.tsx:102]: I checked with ana and this way worked better

File: source/frontend/src/features/documents/documentDetail/DocumentDetailView.tsx (hunk 5 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/documentDetail/DocumentDetailView.tsx:38]: Not sure what you mean.

File: source/frontend/src/components/common/GenericModal.tsx [hunk 7]
```diff
@@ -215,12 +217,19 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
                 title="cancel-modal"
                 variant={cancelButtonVariant ?? 'secondary'}
                 onClick={close}
+                test-id="cancel-modal-button"
```
Comment (asanchezr) [inline source/frontend/src/components/common/GenericModal.tsx:220]: we already have the title property to get the buttons in tests. not opposed to adding test ids but I believe the right syntax is **data-testid="..."** 

See here: https://testing-library.com/docs/queries/bytestid/

File: source/frontend/src/components/common/GenericModal.tsx [hunk 8]
```diff
@@ -215,12 +217,19 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
...
             )}
 
-            <Button title="ok-modal" variant={okButtonVariant ?? 'primary'} onClick={ok}>
+            <Button
+              title="ok-modal"
+              variant={okButtonVariant ?? 'primary'}
+              onClick={ok}
+              disabled={handleOkDisabled}
+              test-id="ok-modal-button"
+            >
```
Comment (asanchezr) [inline source/frontend/src/components/common/GenericModal.tsx:232]: Same comment re: test-id

**Comment by devinleighsmith** [comment]
image

**Comment by devinleighsmith** [comment]
image

**Comment by devinleighsmith** [comment]
@FuriousLlama just trying to save Praveen some time, if Ana and you have already talked and she approved your version of the modal, she needs to update her mockup or Praveen will log bugs stating the mockup doesn't match - lets get ahead of that.

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**